python/xorbits/numpy/numpy_adapters/core.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import inspect
+import warnings
+from typing import Any, Callable, Dict, Type
+
+import numpy as np
+
+from ...core.adapter import MarsOutputType, mars_remote, wrap_mars_callable
+
+_NO_ANNOTATION_FUNCS: Dict[Callable, MarsOutputType] = {
+    np.random.default_rng: MarsOutputType.object,
+    np.random.Generator: MarsOutputType.object,
+    np.random.PCG64: MarsOutputType.object,
+    np.random.MT19937: MarsOutputType.object,
+    np.random.Generator: MarsOutputType.object,
+}
+
+
+def _get_output_type(func: Callable) -> MarsOutputType:
+    try:  # pragma: no cover
+        return_annotation = inspect.signature(func).return_annotation
+        if return_annotation is inspect.Signature.empty:
+            # mostly for python3.7 whose return_annotation is always empty
+            return _NO_ANNOTATION_FUNCS.get(func, MarsOutputType.object)
+        all_types = [t.strip() for t in return_annotation.split("|")]
+
+        return (
+            MarsOutputType.tensor if "ndarray" in all_types else MarsOutputType.object
+        )
+    except (
+        TypeError
+    ):  # some np methods return objects and inspect.signature throws a TypeError
+        return _NO_ANNOTATION_FUNCS.get(func, MarsOutputType.object)
+
+
+def wrap_numpy_module_method(np_cls: Type, func_name: str):
```

Comment:
This function is mostly the same as `wrap_pandas_module_method`. It would be better to extract the common parts and write a util function for easier maintenance in the future.